test/E2ETest/Microsoft.Test.E2E.AspNet.OData/Build.AspNet/BulkOperation/BulkOperationTest.cs

```diff
@@ -0,0 +1,190 @@
+        {
+            //Arrange
+
+            string requestUri = this.BaseAddress + "/convention/Employees";
+
+            var content = @"{
+                    'Name':'SqlUD',
+                    'Friends':[{ 'Id':1001, 'Name' : 'Friend 1001', 'Age': 31},{ 'Id':1002, 'Name' : 'Friend 1002', 'Age': 32},{ 'Id':1003, 'Name' : 'Friend 1003', 'Age': 33}]
+                     }";
+
+            var expectedContext = "{\"@odata.context\":\""+ this.BaseAddress + "/convention/$metadata#Employees/$entity\",\"ID\":0,\"Name\":\"SqlUD\",\"SkillSet\":[],\"Gender\":\"0\",\"AccessLevel\":\"0\",\"FavoriteSports\":null}";
+
+            var requestForPatch = new HttpRequestMessage(new HttpMethod("POST"), requestUri);
+
+            StringContent stringContent = new StringContent(content: content, encoding: Encoding.UTF8, mediaType: "application/json");
+            requestForPatch.Content = stringContent;
+
+            //Act & Assert
+            using (HttpResponseMessage response = await this.Client.SendAsync(requestForPatch))
+            {
+                Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+                var json = response.Content.ReadAsStringAsync().Result;
+                Assert.Contains("SqlUD", json);
+                Assert.Equal(expectedContext.ToString().ToLower(), json.ToString().ToLower());
+            }
+        }
+
+        [Fact]
+        public async Task PostEmployee_WithCreateFriendsFullMetadata()
+        {
+            //Arrange
+
+            string requestUri = this.BaseAddress + "/convention/Employees?$format=application/json;odata.metadata=full";
+
+            string content = @"{
+                    'Name':'SqlUD',
+                    'Friends':[{ 'Id':1001, 'Name' : 'Friend 1001', 'Age': 31},{ 'Id':1002, 'Name' : 'Friend 1002', 'Age': 32},{ 'Id':1003, 'Name' : 'Friend 1003', 'Age': 33}]
+                     }";
+
+            string expectedResponse = "{" +
+                "\"@odata.context\":\"" + this.BaseAddress+ "/convention/$metadata#Employees/$entity\"," +
+                "\"@odata.type\":\"#Microsoft.Test.E2E.AspNet.OData.BulkOperation.Employee\"," +
+                "\"@odata.id\":\"" + this.BaseAddress + "/convention/Employees(0)\"," +
+                "\"@odata.editLink\":\"" + this.BaseAddress + "/convention/Employees(0)\"," +
+                "\"ID\":0,\"Name\":\"SqlUD\"," +
+                "\"SkillSet@odata.type\":\"#Collection(Microsoft.Test.E2E.AspNet.OData.BulkOperation.Skill)\"," +
+                "\"SkillSet\":[],\"Gender@odata.type\":\"#Microsoft.Test.E2E.AspNet.OData.BulkOperation.Gender\"," +
+                "\"Gender\":\"0\"," +
+                "\"AccessLevel@odata.type\":\"#Microsoft.Test.E2E.AspNet.OData.BulkOperation.AccessLevel\"," +
+                "\"AccessLevel\":\"0\"," +
+                "\"FavoriteSports\":null," +
+                "\"Friends@odata.associationLink\":\"" + this.BaseAddress + "/convention/Employees(0)/Friends/$ref\"," +
```

Comment:
Because this is a POST request and we've gotten the logic for serializing a deep insert out of this PR- That's serializing the navigation properties also because there is some redesigning that needs to be done there.